src/main/webapp/app/entities/participation/participation.component.ts

```diff
@@ -70,14 +88,49 @@ export class ParticipationComponent implements OnInit, OnDestroy {
                     this.isLoading = false;
                 });
                 if (this.exercise.type === ExerciseType.PROGRAMMING) {
-                    this.programmingSubmissionService.getSubmissionStateOfExercise(this.exercise.id).subscribe(() => (this.hasLoadedPendingSubmissions = true));
+                    this.programmingSubmissionService
+                        .getSubmissionStateOfExercise(this.exercise.id)
+                        .pipe(
+                            tap((exerciseSubmissionState: ExerciseSubmissionState) => {
+                                this.exerciseSubmissionState = exerciseSubmissionState;
+                            }),
+                        )
+                        .subscribe(() => (this.hasLoadedPendingSubmissions = true));
                 }
                 this.newManualResultAllowed = areManualResultsAllowed(this.exercise);
                 this.presentationScoreEnabled = this.checkPresentationScoreConfig();
             });
         });
     }
 
+    updateParticipationFilter(newValue: FilterProp) {
+        this.isLoading = true;
+        setTimeout(() => {
+            this.participationCriteria.filterProp = newValue;
+            this.isLoading = false;
+        });
+    }
+
+    filterParticipationByProp = (participation: Participation) => {
+        switch (this.participationCriteria.filterProp) {
+            case FilterProp.FAILED:
+                return this.hasFailedSubmission(participation);
+            case FilterProp.ALL:
```

Comment:
I guess you can omit this case as you return true by default anyway. You can even simply this to if else since you have only two cases

Reply:
Hm, when dealing with an enum (especially one that will likely support more than two values in the future), I tend to prefer a switch statement from the beginning to show that each enum value should have its own case. Same reasoning for having the `FilterProp.ALL` case, just to be more explicit. I have combined the last two cases though.